src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_NUMBER = 100;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        FigureType randomFigure = FigureType.values()[random.nextInt(FigureType.values().length)];
+        switch (randomFigure) {
+            case CIRCLE:
+                return new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_NUMBER));
+            case SQUARE:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_NUMBER));
+            case RECTANGLE:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_NUMBER),
+                        random.nextInt(MAX_NUMBER));
+            case RIGHT_RECTANGLE:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_NUMBER),
+                        random.nextInt(MAX_NUMBER));
+            case ISOSCELES_TRAPEZOID:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_NUMBER),
+                        random.nextInt(MAX_NUMBER),
+                        random.nextInt(MAX_NUMBER));
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE,10);
```

Comment:
![image](https://github.com/mate-academy/jv-oop-advanced/assets/73041091/777db0db-74ed-473f-a0be-7735e7b69be5)

Reply:
In this case, I create the default shape Circle (color - white, radius - 10) and in this case this is the case, since I know that I am creating a circle that only has a radius and color. Because in this case, I have to make the color a constant, and when I want to see what kind of circle I’m creating, I’ll have to look for it somewhere above. In your example, this was the case because we define the important variable FIGURE_COUNT, and in this case we are dealing with 1 default figure, and we even know that it will be a circle. I understand that you are trying to teach me the correct code style, but it seems to me that not all “magic numbers” need to be written into a constant, at least to optimize time and not clutter up the class with constants.